library/pkparse.c

```diff
@@ -866,7 +866,7 @@ static int pk_parse_key_sec1_der( mbedtls_ecp_keypair *eck,
     int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
     int version, pubkey_done;
     size_t len;
-    mbedtls_asn1_buf params;
+    mbedtls_asn1_buf params = {};
```

Comment:
```suggestion
    mbedtls_asn1_buf params = { 0, 0, NULL };
```
struct `mbedtls_asn1_buf` has params, which need initialisation. There are multiple other instances in this file also not initialised.